_docs/step-development-guideline.md

```diff
@@ -74,3 +74,36 @@ You should postfix the output ID with `_LIST` (e.g. `OUTPUT_PATH_LIST`), and pro
 ## Version naming convention
 
 You should use [semantic versioning](http://semver.org/) (MAJOR.MINOR.PATCH) for your step. For example: `1.2.3`.
+
+## Step Grouping convention
+
+You can use `project_type_tags` and `type_tags` to group/categorise your steps.
+
+`project_type_tags` are used to control if the step is available/useful for the given project type.
+
+Available `project_type_tags`:
+
+- ios
+- macos
+- android
+- xamarin
+- react-native
+- cordova
+- ionic
+
+_If step is available for all project types, do not specify project_type_tags, otherwise specify every project types, with which the step can work._
+
+`type_tags` are used to categorise the steps based on it's functionality.
+
+Available `type_tags`:
```

Comment:
others can also be used, highlight here that **please include one of these** in the type tags